### DISCUSSION TITLE
New kind of logical operators ( !, &&, ||).

### DISCUSSION THREAD

**Discussion by IgorSivirinov** [dotnet/csharplang#4688] [Language Ideas] [answered]
### **Summary**

Add a new kind of logical operators.
> not it is !
> and it is &&
> or it is ||

```C#
if ((x > y and x < z) or not(y >= v or y >= w))
{
    Console.WriteLine($"Sum: {x + y}");
}
```

### **Motivation**

1. Improve the readability of the code.
2. Words are easier and faster to write than signs.
3. New developers coming from other new programming languages will find it easier to get used to.

**Comment by CyrusNajmabadi** [comment]
These are specifically not expression operators to prevent ambiguity between patterns and expressions. This would break that.

**Comment by HaloFour** [comment, marked as answer]
The `and`, `or` and `not` keywords have already been adopted for use with patterns and cannot be repurposed for aliases of the existing boolean operators without introducing ambiguities.
